src/app/auth.service.spec.ts

```diff
@@ -0,0 +1,120 @@
+import {TestBed, inject} from '@angular/core/testing';
+import {AuthService} from './auth.service';
+
+import {
+  HttpClientTestingModule,
+  HttpTestingController,
+} from '@angular/common/http/testing';
+
+import {AngularFireModule} from '@angular/fire';
+import {auth} from 'firebase/app';
+
+import {Recommendation} from './models/recommendation.model';
+import {environment} from '../environments/environment';
+
+describe('AuthService', () => {
+  let service: AuthService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        HttpClientTestingModule,
+        AngularFireModule.initializeApp(environment.firebaseConfig),
+      ],
+      providers: [AuthService],
+    });
+    service = TestBed.inject(AuthService);
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+
+  it('should make HTTP request for song recommendation', inject(
+    [HttpTestingController, AuthService],
+    (httpMock: HttpTestingController, authService: AuthService) => {
+      const idToken = 'idToken';
+
+      const recommendation: Recommendation = {
+        album: 'album',
+        apple_music_player_url: 'apple_music_player_url',
+        artist: 'artist',
+        embed_content: 'embed_content',
+        id: 0,
+        song_art_image_url: 'song_art_image_url',
+        title: 'title',
+        url: 'url',
+      };
+
+      authService.getRecommendation(idToken).subscribe(response => {
+        expect(response['recommendation']).toEqual(recommendation);
+      });
+
+      const req = httpMock.expectOne(environment.url + '_recommend');
+      expect(req.request.method).toEqual('GET');
+
+      req.flush({recommendation: recommendation});
+    }
+  ));
+
+  it('should sign-in a user and return a credential', done => {
+    const cred = {
+      credential: {
+        idToken: 'idT0ken',
+      },
+      user: {
+        email: 'user@gmail.com',
+      },
+    };
+
+    const mockAngularFireAuth = {
+      auth: jasmine.createSpyObj('auth', {
+        signInWithPopup: Promise.resolve({
+          cred: cred,
+        }),
+      }),
+    };
+
+    mockAngularFireAuth.auth.signInWithPopup().then(data => {
+      expect(data.cred).toEqual(cred);
+      done();
+    });
+  });
+
+  it('should reject accounts existing with different credentials', done => {
+    const code: string = 'auth/account-exists-with-different-credential';
+
+    const mockAngularFireAuth = {
+      auth: jasmine.createSpyObj('auth', {
+        signInWithPopup: Promise.reject({
+          code: code,
+        }),
+      }),
+    };
+
+    mockAngularFireAuth.auth
+      .signInWithPopup()
+      .catch((error: {code: string}) => {
+        expect(error.code).toEqual(code);
+        done();
+      });
+  });
+
+  it('should not have a signed-in user initially', () => {
+    const user = auth().currentUser;
+    expect(user).toBeNull();
+  });
+
+  it('should sign the user out', () => {
+    service.signOut();
```

Comment:
according to the test above, this is doing nothing?

you need to set up the test such that auth().currentUser is _not_ null, and then confirm that auth.signOut() makes it null.

you could alternatively test that your mockAuth.signOut is called, and that window.location.reload() is called.